source/globals.d.ts

```diff
@@ -1,6 +1,20 @@
+type ArgumentTypes<T> = T extends (...args: infer U) => infer R ? U : never;
+
+type FirstArg<T extends any[]> =
+	T extends [infer R, ...any[]] ? R :
+	T extends [] ? undefined :
+	never;
+
+type TailArgs<T extends any[]> = ((...args: T) => any) extends ((
+	_: infer First,
+	...rest: infer Rest
+) => any)
+	? T extends any[] ? Rest : ReadonlyArray<Rest[number]>
+	: [];
+
 declare const browser: AnyObject;
 
-type AnyObject = {[key: string]: any};
+type AnyObject = Record<string, any>;
```

Comment:
In case you weren't aware, there is a `Record` type. We should revisit the `AnyObject`'s usage eventually.

Reply:
Can you add a TODO comment so we don't forget?

Reply:
Done.